offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/components/products/MaintainProductsController.groovy

```diff
@@ -0,0 +1,109 @@
+import life.qbic.business.products.create.CreateProductInput
+
+import life.qbic.datamodel.dtos.business.ProductCategory
+import life.qbic.datamodel.dtos.business.ProductId
+import life.qbic.datamodel.dtos.business.services.DataStorage
+import life.qbic.datamodel.dtos.business.services.PrimaryAnalysis
+import life.qbic.datamodel.dtos.business.services.Product
+import life.qbic.datamodel.dtos.business.services.ProductUnit
+import life.qbic.datamodel.dtos.business.services.ProjectManagement
+import life.qbic.datamodel.dtos.business.services.SecondaryAnalysis
+import life.qbic.datamodel.dtos.business.services.Sequencing
+
+/**
+ * <h1>Controls how the information flows into the use cases {@link life.qbic.business.products.create.CreateProduct} and {@link life.qbic.business.products.archive.ArchiveProduct}</h1>
+ *
+ * <p>This class allows to trigger the use cases and respectively create new products, copy or archive them.</p>
+ *
+ * @since 1.0.0
+ *
+ */
+class MaintainProductsController {
+
+    private final CreateProductInput createProductInput
+    private final ArchiveProductInput archiveProductInput
+    private static final Logging log = Logger.getLogger(this.class)
+
+    MaintainProductsController(CreateProductInput createProductInput,
+                               ArchiveProductInput archiveProductInput){
+        this.createProductInput = createProductInput
+        this.archiveProductInput = archiveProductInput
+    }
+
+    /**
+     * Triggers the creation of a products in the database
+     *
+     * @param category The products category which determines what kind of products is created
+     * @param description The description of the products
+     * @param name The name of the products
+     * @param unitPrice The unit price of the products
+     * @param unit The unit in which the products is measured
+     */
+    void createNewProduct(ProductCategory category, String description, String name, double unitPrice, ProductUnit unit){
+        try {
+            Product product = ProductConverter.createProduct(category, description, name, unitPrice, unit)
+            createProductInput.create(product)
+        } catch (Exception unexpected) {
+            log.error("unexpected exception during create product call", unexpected)
+            throw new IllegalArgumentException("Could not create products from provided arguments.")
+        }
+    }
+
+    /**
+     * Triggers the archiving of a product
+     * @param productId The ProductId of the product that should be archived
+     */
+    void archiveProduct(ProductId productId){
+        try{
+            archiveProductInput.archive(productId)
+        }catch(Exception unexpected){
+            log.error("unexpected exception at archive product call", unexpected)
+            throw new IllegalArgumentException("Could not create products from provided arguments.")
+        }
+    }
+
+    private static class ProductConverter{
```

Comment:
Why is this class a "Converter"? Shouldn't it be a generator or builder since it creates a new object instead of converting something?

Reply:
I think it only packages information, correct @jenniferboedker ?

Reply:
Right, I convert the information for the packages into package DTOs.